src/scipp/compat/xarray_compat.py

```diff
@@ -79,6 +123,46 @@ def from_xarray_dataarray(da: xr.DataArray) -> DataArray:
                      name=da.name or "")
 
 
+def to_xarray_dataarray(da: DataArray) -> xr.DataArray:
+    """Converts a scipp.DataArray object to an xarray.DataArray object.
+
+    Parameters
+    ----------
+    da:
+        A DataArray object to be converted.
+
+    Returns
+    -------
+    :
+        The converted xarray.DataArray object.
+
+    See Also
+    --------
+    scipp.compat.to_xarray_dataarray, scipp.compat.from_xarray_dataset
+    """
+
+    import xarray as xr
+    data = _var_to_xarray(da.data)
+    coords = {}
+    for key, coord in da.coords.items():
+        if da.coords.is_edges(key):
+            raise ValueError("Xarray does not support coordinates with bin edges.")
+        coords[key] = _var_to_xarray(coord)
+
+    out = xr.DataArray(data=data['values'],
+                       dims=data['dims'],
+                       coords={key: coord['values']
+                               for key, coord in coords.items()})
```

Comment:
Why are dims for coords not given/used? I suppose it can be guess for dimension-coordinates, but what about other cases?